[netcdf-c] Build failure

Build fails with the following error


"CMake Error in liblib/CMakeLists.txt:
  Target "netcdf" INTERFACE_INCLUDE_DIRECTORIES property contains path:

    "D:/vcpkg/buildtrees/netcdf-c/src/83aa5b8e47-aa6dbcacd4.clean"

  which is prefixed in the source directory."

The installation of netcdf-c failed. I hope this helps
[config-x64-windows-dbg-err.log](https://github.com/microsoft/vcpkg/files/7609365/config-x64-windows-dbg-err.log)
[config-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/7609367/config-x64-windows-dbg-out.log)

Console output:
"
D:\vcpkg install netcdf-c[hdf5,netcdf-4,platform-default-features]
Computing installation plan...
The following packages will be built and installed:
    netcdf-c[core,dap,hdf5,nczarr,netcdf-4,platform-default-features]:x64-windows -> 4.8.1
Detecting compiler hash for triplet x64-windows...
Restored 0 packages from C:\Users\Gustavo\AppData\Local\vcpkg\archives in 133.6 us. Use --debug to see more details.
Starting package 1/1: netcdf-c:x64-windows
Building package netcdf-c[core,dap,hdf5,nczarr,netcdf-4,platform-default-features]:x64-windows...
-- Using cached Unidata-netcdf-c-cd6173f472b778fa0e558982c59f7183aa5b8e47.tar.gz.
-- Cleaning sources at D:/vcpkg/buildtrees/netcdf-c/src/83aa5b8e47-aa6dbcacd4.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source D:/vcpkg/downloads/Unidata-netcdf-c-cd6173f472b778fa0e558982c59f7183aa5b8e47.tar.gz
-- Applying patch no-install-deps.patch
-- Applying patch fix-dependency-zlib.patch
-- Applying patch use_targets.patch
-- Applying patch fix-dependency-libmath.patch
-- Applying patch fix-linkage-error.patch
-- Applying patch fix-pkgconfig.patch
-- Applying patch fix-manpage-msys.patch
-- Applying patch fix-dependency-libzip.patch
-- Using source at D:/vcpkg/buildtrees/netcdf-c/src/83aa5b8e47-aa6dbcacd4.clean
-- Using cached msys-mingw-w64-i686-pkg-config-0.29.2-2-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-i686-libwinpthread-git-8.0.0.5906.c9a21571-1-any.pkg.tar.zst.
-- Using msys root at D:/vcpkg/downloads/tools/msys2/aa5af7b2aa7e90e8
-- Found external ninja('1.10.2').
-- Configuring x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:127 (message):
    Command failed: "C:/Program Files/CMake/bin/cmake.exe" D:/vcpkg/buildtrees/netcdf-c/src/83aa5b8e47-aa6dbcacd4.clean -DBUILD_TESTING=OFF -DENABLE_EXAMPLES=OFF -DENABLE_TESTS=OFF -DENABLE_FILTER_TESTING=OFF -DENABLE_DAP_REMOTE_TESTS=OFF -DDISABLE_INSTALL_DEPENDENCIES=ON -DNC_USE_STATIC_CRT=OFF -DENABLE_DAP=ON -DENABLE_NETCDF_4=ON -DENABLE_HDF5=ON -DENABLE_NCZARR=ON -DENABLE_NCZARR_ZIP=OFF -DBUILD_UTILITIES=OFF "-DCMAKE_MAKE_PROGRAM=C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" -DBUILD_SHARED_LIBS=ON -DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=D:/vcpkg/scripts/toolchains/windows.cmake -DVCPKG_TARGET_TRIPLET=x64-windows -DVCPKG_SET_CHARSET_FLAG=ON -DVCPKG_PLATFORM_TOOLSET=v142 -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON -DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE -DCMAKE_VERBOSE_MAKEFILE=ON -DVCPKG_APPLOCAL_DEPS=OFF -DCMAKE_TOOLCHAIN_FILE=D:/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON -DVCPKG_CXX_FLAGS= -DVCPKG_CXX_FLAGS_RELEASE= -DVCPKG_CXX_FLAGS_DEBUG= -DVCPKG_C_FLAGS= -DVCPKG_C_FLAGS_RELEASE= -DVCPKG_C_FLAGS_DEBUG= -DVCPKG_CRT_LINKAGE=dynamic -DVCPKG_LINKER_FLAGS= -DVCPKG_LINKER_FLAGS_RELEASE= -DVCPKG_LINKER_FLAGS_DEBUG= -DVCPKG_TARGET_ARCHITECTURE=x64 -DCMAKE_INSTALL_LIBDIR:STRING=lib -DCMAKE_INSTALL_BINDIR:STRING=bin -D_VCPKG_ROOT_DIR=D:/vcpkg -D_VCPKG_INSTALLED_DIR=D:/vcpkg/installed -DVCPKG_MANIFEST_INSTALL=OFF -G Ninja -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=D:/vcpkg/packages/netcdf-c_x64-windows/debug
    Working Directory: D:/vcpkg/buildtrees/netcdf-c/x64-windows-dbg
    Error code: 1
    See logs for more information:
      D:\vcpkg\buildtrees\netcdf-c\config-x64-windows-dbg-out.log
      D:\vcpkg\buildtrees\netcdf-c\config-x64-windows-dbg-err.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:423 (vcpkg_execute_required_process)
  ports/netcdf-c/portfile.cmake:62 (vcpkg_cmake_configure)
  scripts/ports.cmake:142 (include)


Error: Building package netcdf-c:x64-windows failed with: BUILD_FAILED

"

I have got the same error with the latest version:


 D:\vcpkg> .\vcpkg install netcdf-c
Computing installation plan...
The following packages will be built and installed:
    netcdf-c[core,dap,hdf5,nczarr,netcdf-4,platform-default-features]:x64-windows -> 4.8.1#1
Detecting compiler hash for triplet x64-windows...
Restored 0 packages from C:\Users\Gustavo\AppData\Local\vcpkg\archives in 588.5 us. Use --debug to see more details.
Starting package 1/1: netcdf-c:x64-windows
Building package netcdf-c[core,dap,hdf5,nczarr,netcdf-4,platform-default-features]:x64-windows...
-- Using cached Unidata-netcdf-c-cd6173f472b778fa0e558982c59f7183aa5b8e47.tar.gz.
-- Extracting source D:/vcpkg/downloads/Unidata-netcdf-c-cd6173f472b778fa0e558982c59f7183aa5b8e47.tar.gz
-- Applying patch no-install-deps.patch
-- Applying patch fix-dependency-zlib.patch
-- Applying patch use_targets.patch
-- Applying patch fix-dependency-libmath.patch
-- Applying patch fix-linkage-error.patch
-- Applying patch fix-pkgconfig.patch
-- Applying patch fix-manpage-msys.patch
-- Applying patch fix-dependency-libzip.patch
-- Using source at D:/vcpkg/buildtrees/netcdf-c/src/83aa5b8e47-aa6dbcacd4.clean
-- Using cached msys-mingw-w64-i686-pkg-config-0.29.2-2-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-i686-libwinpthread-git-8.0.0.5906.c9a21571-1-any.pkg.tar.zst.
-- Using msys root at D:/vcpkg/downloads/tools/msys2/aa5af7b2aa7e90e8
-- Found external ninja('1.10.2').
-- Configuring x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:127 (message):
    Command failed: "C:/Program Files/CMake/bin/cmake.exe" D:/vcpkg/buildtrees/netcdf-c/src/83aa5b8e47-aa6dbcacd4.clean -DBUILD_TESTING=OFF -DENABLE_EXAMPLES=OFF -DENABLE_TESTS=OFF -DENABLE_FILTER_TESTING=OFF -DENABLE_DAP_REMOTE_TESTS=OFF -DDISABLE_INSTALL_DEPENDENCIES=ON -DNC_USE_STATIC_CRT=OFF -DENABLE_DAP=ON -DENABLE_NETCDF_4=ON -DENABLE_HDF5=ON -DENABLE_NCZARR=ON -DENABLE_NCZARR_ZIP=OFF -DBUILD_UTILITIES=OFF "-DCMAKE_MAKE_PROGRAM=C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" -DBUILD_SHARED_LIBS=ON -DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=D:/vcpkg/scripts/toolchains/windows.cmake -DVCPKG_TARGET_TRIPLET=x64-windows -DVCPKG_SET_CHARSET_FLAG=ON -DVCPKG_PLATFORM_TOOLSET=v142 -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON -DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE -DCMAKE_VERBOSE_MAKEFILE=ON -DVCPKG_APPLOCAL_DEPS=OFF -DCMAKE_TOOLCHAIN_FILE=D:/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON -DVCPKG_CXX_FLAGS= -DVCPKG_CXX_FLAGS_RELEASE= -DVCPKG_CXX_FLAGS_DEBUG= -DVCPKG_C_FLAGS= -DVCPKG_C_FLAGS_RELEASE= -DVCPKG_C_FLAGS_DEBUG= -DVCPKG_CRT_LINKAGE=dynamic -DVCPKG_LINKER_FLAGS= -DVCPKG_LINKER_FLAGS_RELEASE= -DVCPKG_LINKER_FLAGS_DEBUG= -DVCPKG_TARGET_ARCHITECTURE=x64 -DCMAKE_INSTALL_LIBDIR:STRING=lib -DCMAKE_INSTALL_BINDIR:STRING=bin -D_VCPKG_ROOT_DIR=D:/vcpkg -D_VCPKG_INSTALLED_DIR=D:/vcpkg/installed -DVCPKG_MANIFEST_INSTALL=OFF -G Ninja -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=D:/vcpkg/packages/netcdf-c_x64-windows/debug
    Working Directory: D:/vcpkg/buildtrees/netcdf-c/x64-windows-dbg
    Error code: 1
    See logs for more information:
      D:\vcpkg\buildtrees\netcdf-c\config-x64-windows-dbg-out.log
      D:\vcpkg\buildtrees\netcdf-c\config-x64-windows-dbg-err.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:419 (vcpkg_execute_required_process)
  ports/netcdf-c/portfile.cmake:62 (vcpkg_cmake_configure)
  scripts/ports.cmake:142 (include)


Error: Building package netcdf-c:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `git pull` and `.\vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+netcdf-c
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[netcdf-c]+Build+error
including:

  package: netcdf-c[core,dap,hdf5,nczarr,netcdf-4,platform-default-features]:x64-windows -> 4.8.1#1
    vcpkg-tool version: 2021-12-09-724cea8b33cbf06c645f5095fa29773697da9761
    vcpkg-scripts version: 0fd31b9db 2021-12-20 (4 hours ago)

Additionally, attach any relevant sections from the log files above
[config-x64-windows-dbg-err.log](https://github.com/microsoft/vcpkg/files/7755208/config-x64-windows-dbg-err.log)
[config-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/7755209/config-x64-windows-dbg-out.log)
.

--- a/ports/netcdf-c/fix-dependency-mpi.patch
+++ b/ports/netcdf-c/fix-dependency-mpi.patch
@@ -1,0 +1,11 @@
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -1207,7 +1207,8 @@
+   IF(NOT HDF5_PARALLEL)
+     SET(USE_PARALLEL OFF CACHE BOOL "")
+     MESSAGE(STATUS "Cannot find HDF5 library built with parallel support. Disabling parallel build.")
+   ELSE()
+     FIND_PACKAGE(MPI REQUIRED)
++    list(REMOVE_ITEM MPI_C_INCLUDE_PATH "${CMAKE_CURRENT_SOURCE_DIR}")
+     SET(HDF5_PARALLEL ON CACHE BOOL "")
+     SET(USE_PARALLEL ON CACHE BOOL "")

--- a/ports/netcdf-c/portfile.cmake
+++ b/ports/netcdf-c/portfile.cmake
@@ -13,6 +13,7 @@ vcpkg_from_github(
         fix-pkgconfig.patch
         fix-manpage-msys.patch
         fix-dependency-libzip.patch
+        fix-dependency-mpi.patch
 )
 
 #Remove outdated find modules

--- a/ports/netcdf-c/vcpkg.json
+++ b/ports/netcdf-c/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "netcdf-c",
   "version": "4.8.1",
-  "port-version": 1,
+  "port-version": 2,
   "description": "A set of self-describing, machine-independent data formats that support the creation, access, and sharing of array-oriented scientific data.",
   "homepage": "https://github.com/Unidata/netcdf-c",
   "dependencies": [

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -4698,7 +4698,7 @@
     },
     "netcdf-c": {
       "baseline": "4.8.1",
-      "port-version": 1
+      "port-version": 2
     },
     "netcdf-cxx4": {
       "baseline": "4.3.1",

--- a/versions/n-/netcdf-c.json
+++ b/versions/n-/netcdf-c.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "f4d487e2f7e093b2406887023bb3a0856735bad5",
+      "version": "4.8.1",
+      "port-version": 2
+    },
+    {
       "git-tree": "9d8a0a97fd75563172e3f1947608db3802c3b84a",
       "version": "4.8.1",
       "port-version": 1

